arcgis-ios-sdk-samples/Layers/Apply raster rendering rule/ApplyRasterRenderingRuleViewController.swift

```diff
@@ -0,0 +1,116 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class ApplyRasterRenderingRuleViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap()
+        }
+    }
+    /// The choose rendering rule button.
+    @IBOutlet var chooseRenderingRuleBarButtonItem: UIBarButtonItem!
+    
+    // MARK: Properties and methods
+    
+    /// An online image service that features Charlotte, North Carolina's downtown area.
+    let imageServiceURL = URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/CharlotteLAS/ImageServer")!
+    /// A list of rendering rule info supported by the service.
+    var rasterRenderingRules = [AGSRenderingRuleInfo]()
+    /// A dictionary to cache created raster layers.
+    var rasterLayers = [String: AGSRasterLayer]()
+    
+    /// Create a map.
+    ///
+    /// - Returns: An `AGSMap` object.
+    func makeMap() -> AGSMap {
+        let map = AGSMap(basemap: .streets())
+        // Create a raster layer from the raster and add it to the map.
+        let imageServiceRaster = AGSImageServiceRaster(url: imageServiceURL)
+        map.operationalLayers.add(makeRasterLayer(raster: imageServiceRaster))
+        // Load the raster and get a list of rendering rule info supported by the service.
+        imageServiceRaster.load { [weak self, unowned imageServiceRaster] error in
+            guard let self = self else { return }
+            if let serviceInfo = imageServiceRaster.serviceInfo, let extent = serviceInfo.fullExtent {
+                self.mapView.setViewpoint(AGSViewpoint(targetExtent: extent), completion: nil)
+                self.rasterRenderingRules = serviceInfo.renderingRuleInfos
+                self.chooseRenderingRuleBarButtonItem.isEnabled = true
+            } else if let error = error {
+                self.presentAlert(error: error)
+            }
+        }
+        return map
+    }
+    
+    /// Create a raster layer from an image service raster with an optional rendering rule.
+    ///
+    /// - Parameters:
+    ///   - raster: An `AGSImageServiceRaster` object.
+    ///   - renderingRule: An optional `AGSRenderingRule` object.
+    /// - Returns: An `AGSRasterLayer` object.
+    func makeRasterLayer(raster: AGSImageServiceRaster, renderingRule: AGSRenderingRule? = nil) -> AGSRasterLayer {
+        raster.renderingRule = renderingRule
```

Comment:
Since `renderingRule` is a property of the passed in `raster`, it seems like it should be set by whoever created the raster and passed it into this function in the first place. If it were moved outside of this function, then the only thing left would be `AGSRasterLayer(raster: raster)`, which makes me wonder if this function is worth having. What do you think?

Reply:
😅 😅 That totally makes sense... Removed.